cmake/BLTOptions.cmake

```diff
@@ -61,6 +61,7 @@ option(ENABLE_MPI          "Enables MPI support" OFF)
 option(ENABLE_OPENMP       "Enables OpenMP compiler support" OFF)
 option(ENABLE_CUDA         "Enable CUDA support" OFF)
 
+option(ENABLE_GMOCK        "Enable Google Mock testing support" OFF)
```

Comment:
It mirrors `ENABLE_MPI`, `ENABLE_OPENMP`, and `ENABLE_CUDA`.
It's also grouped with them in `BLTOptions.cmake`

I thought about adding a more descriptive comment that acknowledges the fact that this also screened by `ENABLE_TESTS` -- would that suffice?